lib/segment/src/id_tracker/mutable_id_tracker.rs

```diff
@@ -368,20 +328,94 @@ fn versions_path(segment_path: &Path) -> PathBuf {
     segment_path.join(FILE_VERSIONS)
 }
 
-fn load_mappings(
-    mappings_path: &Path,
-    deleted: &mut BitVec,
-    internal_to_external: &mut Vec<PointIdType>,
-    external_to_internal_num: &mut BTreeMap<u64, PointOffsetType>,
-    external_to_internal_uuid: &mut BTreeMap<Uuid, PointOffsetType>,
+/// Store new mapping changes, appending them to the given file
```

Comment:
The meat for all store/load logic begins here.

It's all separated in functions like:
- store (append) mappings to file
- write mappings to writer
- write single mapping to writer
- load mappings from file
- read mappings from reader
- read single mapping from reader